File: recette/src/app/project.ts
import {Users} from "./users";

export interface Project{
  id: number;
  name: string;
  description: string;
  url: string;
  vav_metier: string;
  vav_orga: string; // Use the User interface for the vav_orga property
  datedajout: Date; // Use the Date type for the datedajout property
  dpt_id: File; // Use the File interface for the dpt_id property
  T_U: File; // Use the File interface for the T_U property
  envi: boolean; // Set the default value of the 'envi' field to false
  intervention: boolean; // Set the default value of the 'intervention' field to false
etat:string;
}

File: demo/src/main/java/com/recetteapp/security/project/project.java
package com.recetteapp.security.project;

import com.recetteapp.security.Domaine.Domaine;
import com.recetteapp.security.file.file;
import com.recetteapp.security.user.User;
import com.recetteapp.security.user.User2;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class project {
    @Id
    @GeneratedValue
    private Integer id;
    private String name ;
    private String description ;

    private String url ;

    private LocalDateTime datedajout ;

    @ManyToOne
    @JoinColumn(name = "dpt_file_id") // foreign key column name
    private file dpt_id;
    @ManyToOne
    @JoinColumn(name = "tu_file_id")
    private file T_U;
    private boolean intervention = false;
    private boolean envi  = false;
    private String etat ;
    @ManyToOne
    @JoinColumn
    private Domaine domaineid;
}
